### ISSUE TITLE
SpringModule doesn't bind Parameterized Types

### ISSUE THREAD

**Issue by twicksell** [spring-projects/spring-guice#14]
SpringModule binds beans for all of their interfaces. However, it does not bind generic interfaces. To AbstractCompleteWiringTests, add the following:
```
	@Test
	public void getParameterizedType() {
	    Parameterized<String> instance = this.injector.getInstance(Key.get(new TypeLiteral<Parameterized<String>>() {}));
	    assertNotNull(instance);
	}

...

	public static interface Parameterized<T> {
	}
```

To SpringModuleWiringTests add the following:
```
		@Bean
		public Parameterized<String> parameterizedBean() {
		    return new Parameterized<String>(){};
		}
```

**Comment by twicksell** [comment]
Proposed fix would be a bit of a refactor, but mostly revolves around moving SpringModule's usage of Class to Type, and doing something like:
```
                for (Type iface : cls.getGenericInterfaces()) {
                    bindConditionally(binder(), name, iface, typeProvider, namedProvider);
                }
```
Working on a PR now, just trying to get past a couple of regressions.

**Comment by dsyer** [comment]
Fixed in c42ba69
